### PR TITLE
excalidraw_export: init at v1.1.0

### PR DESCRIPTION
**Pull request by venikx**
## Description of changes

The following npm package makes it possible to convert `.excalidraw` files to `.svg` files: https://github.com/Timmmm/excalidraw_export and is used by https://github.com/wdavew/org-excalidraw to handle excalidraw files inside org-mode in Emacs.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by venikx** [comment]
I wasn't sure which way I should provide this package `buildNpmPackage` or via the `node2nix` helper. I saw from the commits that some packages are being dropped and newer ones are added via `buildNpmPackage`. Is there a preference to use `buildNpmPackage`? If yes, I have no issue converting this.

**Comment by tomodachi94** [comment]
> Is there a preference to using buildNpmPackage? If yes, I have no issue converting this.

Yes, there seems to be consensus among Node-involved Nixpkgs maintainers that we should avoid adding new packages to `nodePackages`.

**Comment by venikx** [comment]
> > Is there a preference to using buildNpmPackage? If yes, I have no issue converting this.
> 
> Yes, there seems to be consensus among Node-involved Nixpkgs maintainers that we should avoid adding new packages to `nodePackages`. (c.f. #229475)

Updated to use `buildNpmPackage` instead of the `node2nix` stuff.

File: pkgs/by-name/ex/excalidraw_export/package.nix
Comment (tomodachi94) [inline pkgs/by-name/ex/excalidraw_export/package.nix:32]:
```suggestion
    description = "CLI to export Excalidraw drawings to SVG and PDF";
```
For consistency, we omit trailing punctuation in the description: https://nixos.org/manual/nixpkgs/unstable/#var-meta-description

This description is otherwise great though :)

**Review by tomodachi94 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ex/excalidraw_export/package.nix
Comment (venikx) [reply pkgs/by-name/ex/excalidraw_export/package.nix:29]: I think it's needed since one of the dependencies depends on some binaries being on built on the system. Without it, the installation seem to fail:

```
npm error /nix/store/8jind2ncd72hlzym9pghjjdj79fvqzv5-bash-5.2p32/bin/sh: line 1: pkg-config: command not found
npm error gyp: Call to 'pkg-config pixman-1 --libs' returned exit status 127 while in binding.gyp. while trying to load binding.gyp
npm error gyp ERR! configure error
npm error gyp ERR! stack Error: `gyp` failed with exit code: 1
npm error gyp ERR! stack at ChildProcess.<anonymous> (/nix/store/vc64zpxndmp3xh9699yb2vpbl81pxbk0-nodejs-20.17.0/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:297:18)
npm error gyp ERR! stack at ChildProcess.emit (node:events:519:28)
npm error gyp ERR! stack at ChildProcess._handle.onexit (node:internal/child_process:294:12)
npm error gyp ERR! System Linux 6.8.10-asahi
npm error gyp ERR! command "/nix/store/vc64zpxndmp3xh9699yb2vpbl81pxbk0-nodejs-20.17.0/bin/node" "/nix/store/vc64zpxndmp3xh9699yb2vpbl81pxbk0-nodejs-20.17.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "configure" "--fallback-to-build" "--update-binary" "--module=/build/source/node_modules/canvas/build/Release/canvas.node" "--module_name=canvas" "--module_path=/build/source/node_modules/canvas/build/Release" "--napi_version=9" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=node-v115" "--nodedir=/nix/store/s6jmwml9fhazzvrva1jhmzmr5qrbbhgd-nodejs-20.17.0-source"
npm error gyp ERR! cwd /build/source/node_modules/canvas
npm error gyp ERR! node -v v20.17.0
npm error gyp ERR! node-gyp -v v10.1.0
npm error gyp ERR! not ok
npm error node-pre-gyp ERR! build error
npm error node-pre-gyp ERR! stack Error: Failed to execute '/nix/store/vc64zpxndmp3xh9699yb2vpbl81pxbk0-nodejs-20.17.0/bin/node /nix/store/vc64zpxndmp3xh9699yb2vpbl81pxbk0-nodejs-20.17.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --update-binary --module=/build/source/node_modules/canvas/build/Release/canvas.node --module_name=canvas --module_path=/build/source/node_modules/canvas/build/Release --napi_version=9 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v115 --nodedir=/nix/store/s6jmwml9fhazzvrva1jhmzmr5qrbbhgd-nodejs-20.17.0-source' (1)
npm error node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/build/source/node_modules/@mapbox/node-pre-gyp/lib/util/compile.js:89:23)
npm error node-pre-gyp ERR! stack     at ChildProcess.emit (node:events:519:28)
npm error node-pre-gyp ERR! stack     at maybeClose (node:internal/child_process:1105:16)
npm error node-pre-gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:305:5)
npm error node-pre-gyp ERR! System Linux 6.8.10-asahi
npm error node-pre-gyp ERR! command "/nix/store/vc64zpxndmp3xh9699yb2vpbl81pxbk0-nodejs-20.17.0/bin/node" "/build/source/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build" "--update-binary"
npm error node-pre-gyp ERR! cwd /build/source/node_modules/canvas
npm error node-pre-gyp ERR! node -v v20.17.0
npm error node-pre-gyp ERR! node-pre-gyp -v v1.0.11
npm error node-pre-gyp ERR! not ok
npm error Log files were not written due to an error writing to the directory: /nix/store/31h9lfa8p68kgd9ahn36s1gzcvgz0fjw-excalidraw_export-1.1.0-npm-deps/_logs
npm error You can rerun the command with `--loglevel=verbose` to see the logs in your terminal
error: builder for '/nix/store/n2ww9gn8928b8gknb3vmay9amzc5rsxj-excalidraw_export-1.1.0.drv' failed with exit code 1;
       last 10 log lines:
       > npm error node-pre-gyp ERR! stack     at maybeClose (node:internal/child_process:1105:16)
       > npm error node-pre-gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:305:5)
       > npm error node-pre-gyp ERR! System Linux 6.8.10-asahi
       > npm error node-pre-gyp ERR! command "/nix/store/vc64zpxndmp3xh9699yb2vpbl81pxbk0-nodejs-20.17.0/bin/node" "/build/source/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build" "--update-binary"
       > npm error node-pre-gyp ERR! cwd /build/source/node_modules/canvas
       > npm error node-pre-gyp ERR! node -v v20.17.0
       > npm error node-pre-gyp ERR! node-pre-gyp -v v1.0.11
       > npm error node-pre-gyp ERR! not ok
       > npm error Log files were not written due to an error writing to the directory: /nix/store/31h9lfa8p68kgd9ahn36s1gzcvgz0fjw-excalidraw_export-1.1.0-npm-deps/_logs
       > npm error You can rerun the command with `--loglevel=verbose` to see the logs in your terminal
       For full logs, run 'nix log /nix/store/n2ww9gn8928b8gknb3vmay9amzc5rsxj-excalidraw_export-1.1.0.drv'.
```

**Review by tomodachi94 — APPROVED** [review]
Lgtm, thank you ❤️

File: pkgs/by-name/ex/excalidraw_export/package.nix
Comment (venikx) [inline pkgs/by-name/ex/excalidraw_export/package.nix:36]: I noticed the darwin builds are failing, and I don't have a darwin machine. I guess it would be better practice to limit the platforms?

```suggestion
    mainProgram = "excalidraw_export";
        platforms = [
      "aarch64-linux"
      "x86_64-linux"
    ];
```

Comment (tomodachi94) [reply pkgs/by-name/ex/excalidraw_export/package.nix:36]: That would work. If Darwin builds are supported upstream, but are broken locally (at the Nixpkgs level), setting `broken` to `stdenv.isDarwin` is better.

Comment (venikx) [reply pkgs/by-name/ex/excalidraw_export/package.nix:36]: > That would work, but if Darwin builds are supported upstream, but are broken locally (at the Nixpkgs level), setting `broken` to `stdenv.isDarwin` is better.

I am not sure if I fully understand this. With upstream, do you mean the other dependencies original library? If yes, from what I understand, macOS should be supported by it, but I think it needs to be built differently. The last thing you mentioned, would look like this?

```suggestion
    mainProgram = "excalidraw_export";
    broken = stdenv.isDarwin; 
```

Comment (venikx) [reply pkgs/by-name/ex/excalidraw_export/package.nix:37]: added this broken flag

**Comment by venikx** [comment]
I was able to also verify the build on my desktop linux, so `x86_64-linux` and `aarch64-linux` should both work. Flagged anything darwin related as broken, due to earlier automated builds failing.

**Review by tomodachi94 — APPROVED** [review]
Lgtm, thank you for your patience! 💖

**Comment by venikx** [comment]
@tomodachi94 I'm a bit unfamiliar with the process since it's my first time contributing. Is there something still that is blocking the merge that I can help resolve?

**Comment by tomodachi94** [comment]
Nope, just my forgetfulness :) Merging now, thank you ❤️

**Pull request merged**
